cmake/DetectSystemInformation.cmake

```diff
@@ -26,21 +26,25 @@
 macro(omr_detect_system_information) 
 
    if(CMAKE_SYSTEM_PROCESSOR MATCHES "x86_64|AMD64")
+      set(OMR_HOST_ARCH "x86")
       set(OMR_ARCH_X86 ON)
       set(OMR_ENV_DATA64 ON)
       set(OMR_TARGET_DATASIZE 64)
       set(OMR_ENV_LITTLE_ENDIAN ON)
    elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "x86|i[3-6]86")
+      set(OMR_HOST_ARCH "x86")
       set(OMR_ARCH_X86 ON)
       set(OMR_ENV_DATA32 ON)
       set(OMR_TARGET_DATASIZE 32)
       set(OMR_ENV_LITTLE_ENDIAN ON)
    elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "arm")
+      set(OMR_HOST_ARCH "arm")
       set(OMR_ARCH_ARM ON)
       set(OMR_ENV_DATA32 ON)
       set(OMR_TARGET_DATASIZE 32)
       set(OMR_ENV_LITTLE_ENDIAN ON)
    elseif(CMAKE_SYSTEM_NAME MATCHES "OS390")
+      set(OMR_HOST_ARCH "z")
       set(OMR_ARCH_S390 ON)
       set(OMR_ENV_DATA32 ON)
       set(OMR_TARGET_DATASIZE 32)
```

Comment:
I think I want to keep this macro as a variable exporter that others can depend on.